### DISCUSSION TITLE
GraalVM roadmap

### DISCUSSION THREAD

**Discussion by eruiz** [ballerina-platform/ballerina-lang#32550] [General]
Hi there!

Are there any roadmap about the migration to GraalVM?

Thanks!

**Comment by ThisaruGuruge** [comment]
@warunalakshitha @sameerajayasoma

**Comment by sameerajayasoma** [comment]
@eruiz  We don't have immediate plans to support generating GraalVM native executables right now. We started this twice but had to postpone due to other priorities. I will update this thread once we have some news

**Comment by keizer619** [comment]
@TharmiganK

**Comment by TharmiganK** [comment]
Hi @eruiz 

From Ballerina 2201.7.0 (SwanLake) onwards, Ballerina supports GraalVM AOT compilation to generate standalone executables by passing the graalvm flag in the build command: `bal build --graalvm`.  All Ballerina official libraries(under `ballerina` organization) are GraalVM compatible.

Please check our learn pages for more information: [GraalVM executable overview](https://ballerina.io/learn/graalvm-executable-overview/)

Medium articles on Ballerina GraalVM support:
https://medium.com/ballerina-techblog/ballerina-meets-graalvm-from-code-to-native-executable-6c99b47f6529
https://medium.com/ballerina-techblog/ballerina-meets-graalvm-in-the-cloud-a-winning-formula-for-cloud-native-ed19d79b5fe2
https://medium.com/ballerina-techblog/unveiling-ballerina-graalvm-image-tackling-production-issues-c21798e165c3
